main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -1107,8 +1108,23 @@ object Namer {
     case WeededAst.Pattern.Tag(qname, pat, loc) =>
       NamedAst.Pattern.Tag(qname, visitPattern(pat), loc)
 
-    case WeededAst.Pattern.Tuple(elms, loc) => NamedAst.Pattern.Tuple(elms map visitPattern, loc)
+    case WeededAst.Pattern.Tuple(elms, loc) =>
+      NamedAst.Pattern.Tuple(elms map visitPattern, loc)
 
+    case WeededAst.Pattern.Record(pats, pat, loc) =>
+      val psVal = pats map {
+        case WeededAst.Pattern.Record.RecordFieldPattern(field, None, loc1) =>
+          // Introduce new symbols if there is no pattern
+          val sym = Symbol.freshVarSym(field.name, BoundBy.Pattern, field.loc)
+          val p = NamedAst.Pattern.Var(sym, field.loc)
+          NamedAst.Pattern.Record.RecordFieldPattern(field, p, loc1)
+
+        case WeededAst.Pattern.Record.RecordFieldPattern(field, Some(pat1), loc1) =>
+          NamedAst.Pattern.Record.RecordFieldPattern(field, visitPattern(pat1), loc1)
```

Comment:
Rewrite these two cases to avoid a deep match. You can just match ones and then again on `rest`